### ISSUE TITLE
systemd-resolve www.dnssec-bogus.sg should fail and it's not

### ISSUE THREAD

**Issue by bigon** [systemd/systemd#8901]
Hello,
www.dnssec-bogus.sg is a DNS name with an on purpose boggus DNSSEC signature.

When using bind with DNSSEC validation enabled as a recursive DNS server, I can see a SERVFAIL error as expected.

When using resolved (pointing to a server recursive server NOT doing DNSSEC validation) I have no mention that the signature is invalid.

resolved is configured with `DNSSEC=allow-downgrade`

[log2.txt](https://github.com/systemd/systemd/files/1971856/log2.txt)

**Comment by bigon** [comment]
Worst part it says `Data is authenticated: yes`

```
www.dnssec-bogus.sg: 123.100.252.181

-- Information acquired via protocol DNS in 81.1ms.
-- Data is authenticated: yes
```

For info:

```
$ dig www.dnssec-bogus.sg +dnssec

; <<>> DiG 9.11.3-1-Debian <<>> www.dnssec-bogus.sg +dnssec
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60987
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 2, ADDITIONAL: 9

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 1280
;; QUESTION SECTION:
;www.dnssec-bogus.sg.		IN	A

;; ANSWER SECTION:
www.dnssec-bogus.sg.	300	IN	A	123.100.252.181
www.dnssec-bogus.sg.	300	IN	RRSIG	A 13 3 300 20180508101147 20180506081147 35273 dnssec-bogus.sg. Mu6WY/rkAythMrXinDqf+OuJ281795NLJufjRBN775Zn6aTe2IuQdh1x 7yhy9noYbjL3HzsVqeQHJFj5zRxStg==

;; AUTHORITY SECTION:
dnssec-bogus.sg.	600	IN	NS	duke.ns.cloudflare.com.
dnssec-bogus.sg.	600	IN	NS	sue.ns.cloudflare.com.

;; ADDITIONAL SECTION:
sue.ns.cloudflare.com.	27725	IN	A	173.245.58.145
sue.ns.cloudflare.com.	28016	IN	AAAA	2400:cb00:2049:1::adf5:3a91
duke.ns.cloudflare.com.	810	IN	A	173.245.59.110
duke.ns.cloudflare.com.	31998	IN	AAAA	2400:cb00:2049:1::adf5:3b6e
sue.ns.cloudflare.com.	27725	IN	RRSIG	A 13 4 86400 20180507175352 20180505155352 35273 cloudflare.com. /MQHhbcw3qB99CJNdC1mQPeQI3DiODLvvgG4UJs5OQKRVAuDJtn821tt ckAHKuK4LFw51ZFHa813n9qR4c/MGw==
sue.ns.cloudflare.com.	28016	IN	RRSIG	AAAA 13 4 86400 20180507175843 20180505155843 35273 cloudflare.com. OFqudGnsvm7JlsT8N9e5gpKlHCbxLULv3lQ3cm3AD10wV1behIBLNO2w bwJSmQPBaCbhMHTd2+ef6XI8nA7Qxw==
duke.ns.cloudflare.com.	810	IN	RRSIG	A 13 4 86400 20180507102517 20180505082517 35273 cloudflare.com. wTSjt+MkFC9agFooAz+kY5QYgWRdleSW5j6DvtE+J897IwIsZ18UzYQv Wkopj06PvubTm9tptn4Y/EP7GBBBWg==
duke.ns.cloudflare.com.	31998	IN	RRSIG	AAAA 13 4 86400 20180507190505 20180505170505 35273 cloudflare.com. yqPNF7/yTUkD4noPkXbYHstCdkjKr8tELMcl0MFzqDzdhyfln7nUIL7S U3tfD5/MtaDnDzWmmqo/NapJbN7JDA==

;; Query time: 15 msec
;; SERVER: 10.200.0.200#53(10.x.x.x)
;; WHEN: Mon May 07 11:11:47 CEST 2018
;; MSG SIZE  rcvd: 757
```

**Comment by poettering** [comment]
Fix in #9240
